plugins/scaffolder-backend-module-sentry/src/actions/createProject.test.ts

```diff
@@ -0,0 +1,222 @@
+/*
+ * Copyright 2021 The Backstage Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import { ConfigReader } from '@backstage/config';
+import { JsonObject } from '@backstage/types';
+import { createSentryCreateProjectAction } from './createProject';
+import { ActionContext } from '@backstage/plugin-scaffolder-node';
+import { InputError } from '@backstage/errors';
+import { randomBytes } from 'crypto';
+
+describe('sentry:project:create action', () => {
+  const createScaffolderConfig = (configData: JsonObject = {}) => ({
+    config: new ConfigReader({
+      scaffolder: {
+        ...configData,
+      },
+    }),
+  });
+
+  let fetch: jest.Func;
+
+  const mockFetch = (response = {}) => {
+    const mockedResponse = {
+      status: 201,
+      headers: {
+        get: () => 'application/json',
+      },
+      json: async () =>
+        Promise.resolve({
+          detail: 'project creation mocked result',
+        }),
+      text: async () => Promise.resolve('Unexpected error.'),
+      ...response,
+    };
+
+    fetch = jest.fn().mockImplementation(() => Promise.resolve(mockedResponse));
```

Comment:
Mocking fetch is a bit unfortunate for a couple of reasons, one of which being that the Response object is a bit of a pain to deal with and fake.

It's pretty established within the project to lean heavily on the `msw` library for network mocking. See for example [what the confluence module does](https://github.com/backstage/backstage/blob/1b7d382f8f4a79e357f769b676f9b172dd342e94/plugins/scaffolder-backend-module-confluence-to-markdown/src/actions/confluence/confluenceToMarkdown.examples.test.ts#L136) in its tests. The benefit is that you detach yourself from worrying about how fetch works, and you have this entire nice platform to leverage for how to fake various network calls. As a bonus, when using the [`setupRequestMockHandlers`](https://github.com/backstage/backstage/blob/1b7d382f8f4a79e357f769b676f9b172dd342e94/plugins/scaffolder-backend-module-confluence-to-markdown/src/actions/confluence/confluenceToMarkdown.examples.test.ts#L47) helper, you ensure that no traffic in the test can ever accidentally leave the test worker.